### PR TITLE
Adding ability to pass an array of servers for implemention of domain sharding

### PR DESCRIPTION
**Pull request by beenanner**

### CODE REVIEW

**Comment by pda** [comment]
I'm thinking selection from a pool of multiple servers is something that can and should be done outside of thumbor; it's unlikely that one selection algorithm or hash function is suitable for everybody, and the interface remains simpler if it takes a single host string.

Also, I don't think adequately even distribution amongst the list of servers will be achieved with `strlen($imgPath) % count($server)` standing in for a real hash function. I think this highlights that the selection strategy should be external.

So, while contributions are always appreciated, I think this one should probably be closed and handled outside phumbor. Anybody agree/disagree?

File: lib/Thumbor/Url.php [hunk 1]
```diff
@@ -44,6 +44,11 @@ public function stringify($server, $secret, $original, $commands)
             $imgPath = $original;
         }
 
+        if(is_array($server)) {
+            $index = strlen($imgPath) % count($server);
```
Comment (Vektah) [inline lib/Thumbor/Url.php]:
```php
crc32($imgPath) & count($server)
```

Would have a much better distribution

Comment (beenanner) [inline lib/Thumbor/Url.php]: I can implement this pending the answers to @pda question about whether this should be implemented in the bundle.

Comment (navitronic) [inline lib/Thumbor/Url.php]: I personally don't think that this belongs within this library.

A fairly small wrapper class containing an array of servers and a method to map `$original` to server could probably suffice. Maybe something along the lines of:

```php
<?php

use Thumbor\Url;

class PhumborServerFarm
{
    /** @var array */
    private $servers;

    /**
     * @param array $servers
     * @param string $secret
     */
    public function __construct($servers, $secret)
    {
        $this->servers = $servers;
    }

    /**
     * @param string $original
     * @param array $commands
     */
    public function stringify( $original, $commands)
    {
        return Url::stringify($this->getServer($original), $secret, $original, $commands);
    }

    private function getServer($original)
    {
        // return server here
    }
}
```

**Comment by beenanner** [comment]
Thanks everyone on the feedback. All valid points that I can't really disagree with, so will close this out. Thank you all as well for all the work done on this bundle, amazing work!

**Pull request closed without merging**